Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -74,14 +76,36 @@ protected override void Started(EntityUid uid, RevolutionaryRuleComponent compon
     protected override void ActiveTick(EntityUid uid, RevolutionaryRuleComponent component, GameRuleComponent gameRule, float frameTime)
     {
         base.ActiveTick(uid, component, gameRule, frameTime);
-        if (component.CommandCheck <= _timing.CurTime)
+
+        if (component.HeadRevsChosen)
         {
-            component.CommandCheck = _timing.CurTime + component.TimerWait;
+            if (component.CommandCheck <= _timing.CurTime)
+            {
+                component.CommandCheck = _timing.CurTime + component.TimerWait;
+
+                if (CheckCommandLose())
+                {
+                    _roundEnd.DoRoundEndBehavior(RoundEndBehavior.ShuttleCall, component.ShuttleCallTime);
+                    GameTicker.EndGameRule(uid, gameRule);
+                }
+            }
+            return;
+        }
 
-            if (CheckCommandLose())
+        if (component.StartTime != null && component.StartTime <= _timing.CurTime)
+        {
+            var query = QueryActiveRules();
+            while (query.MoveNext(out _, out var comp, out _))
             {
-                _roundEnd.DoRoundEndBehavior(RoundEndBehavior.ShuttleCall, component.ShuttleCallTime);
-                GameTicker.EndGameRule(uid, gameRule);
+                _antagSelection.EligiblePlayers(comp.HeadRevPrototypeId, comp.MaxHeadRevs, comp.PlayersPerHeadRev, comp.HeadRevStartSound,
+                    "head-rev-role-greeting", "#5e9cff", out var chosen);
+                if (chosen.Any())
+                    GiveHeadRev(chosen, comp.HeadRevPrototypeId, comp);
+                else
+                {
+                    _chatManager.SendAdminAnnouncement(Loc.GetString("rev-no-heads"));
+                    comp.HeadRevsChosen = true;
+                }
```

Comment:
Extremely weird code flow where you set headrevs outside the method but also inside the method.

Also you can just remove headrevschosen and just use the start time anyway as they are both functioning similarly.

Reply:
I have headrevschosen so this antag selection code doesn't trigger repeatedly as it's used above at the command check with a return. If I had both on start time I think only the command check would trigger as it wouldn't reach the antag selection.